src/components/deployments/add-scaler/AddScalerForm.jsx

```diff
@@ -0,0 +1,259 @@
+import React from 'react';
+import { Modal, Form, Radio, Input, Alert, Select, Row, Col, Button } from 'antd';
+import RadioCards from '../../radio-cards/RadioCards';
+import FormItemLabel from '../../form-item-label/FormItemLabel';
+import ConditionalFormBlock from '../../conditional-form-block/ConditionalFormBlock';
+import { KedaTiggerType, spaceCloudClusterOrigin } from '../../../constants';
+import { DeleteOutlined, PlusOutlined } from '@ant-design/icons';
+import ObjectAutoComplete from '../../object-autocomplete/ObjectAutoComplete';
+
+const AddScalerForm = (props) =>{
+
+  const { initialValues, secrets, handleSubmit, handleCancel } = props;
+  const [form] = Form.useForm();
+
+  const selectedType = (type) => {
+    switch (type) {
+      case 'requests-per-second':
+        return 'requests-per-second';
+        break;
+      case 'active-requests':
+        return 'active-requests';
+        break;
+      case 'cpu':
+        return 'cpu';
+        break;
+      case 'ram':
+        return 'ram';
+        break;
+      default: return 'keda'
+        break;
+    }
+  }
+
+  const formInitialValue = {
+    name: initialValues && initialValues.name ? initialValues.name : '',
+    type: initialValues && initialValues.type ? selectedType(initialValues.type) : 'requests-per-second',
+    target: initialValues && initialValues.metadata ? initialValues.metadata : '',
+    kedaType: initialValues && initialValues.type ? initialValues.type : null,
+    kedaTargets: initialValues && initialValues.metadata ? initialValues.metadata : [],
+    kedaSecrets: initialValues && initialValues.authRef ? initialValues.authRef : []
+  }
+
+  const handleSubmitClick = () => {
+    form.validateFields().then(values =>{
+      const operation = initialValues ? 'edit' : 'add';
+      values.target = Number(values.target);
+      handleSubmit(values, operation)
+      handleCancel()
+    })
+  }
+
+  const envSecrets = secrets.filter(secret => secret.type === 'env')
```

Comment:
Filtering of secrets should be done in the page itself. This component should directly get envSecrets in the props.